app/controllers/food_controller.rb

```diff
@@ -10,140 +7,90 @@ class FoodController < Sinatra::Base
 
   get "/foods/users/:slug" do
     @user = User.find_by(slug: params[:slug])
-    @logged_in = logged_in?
-    @current_user = current_user
-    @nav = {:activity => {:status => ""}, :exercise => {:status => ""}, :nutrition => {:status => ""}}
-
     if @user
-      @viewing_own_profile_while_logged_in = viewing_own_profile_while_logged_in?(@user,@current_user)
-      if @viewing_own_profile_while_logged_in
-        @nav[:nutrition][:status] = "active"
+      if viewing_own_profile_while_logged_in?(@user)
         @main_heading = "My Meals"
         @title = "Fitness Tracker - My Meals"
       else
-        @main_heading = "#{first_name(@user.name)}'s Meals"
-        @title = "Fitness Tracker - #{first_name(@user.name)}'s Meals"
+        @main_heading = "#{@user.first_name}'s Meals"
+        @title = "Fitness Tracker - #{@user.first_name}'s Meals"
       end
       erb :'foods/index'
     else
-      @title = "Fitness Tracker - Error"
-      flash[:error] = "The user you are looking for doesn't exist."
-      status 404
-      body(erb :error)
+      display_err_page(404, "The user you are looking for doesn't exist.")
     end
   end
 
   get "/foods/new" do
-    @logged_in = logged_in?
-    if @logged_in
-      @current_user = current_user
+    if logged_in?
       @title = "Fitness Tracker - Add Meal"
-      @nav = {:activity => {:status => ""}, :exercise => {:status => ""}, :nutrition => {:status => "active"}}
       erb :'foods/new'
     else
       redirect '/'
     end
   end
 
   post "/foods" do
-    @logged_in = logged_in?
-    @nav = {:activity => {:status => ""}, :exercise => {:status => ""}, :nutrition => {:status => "active"}}
-    if @logged_in
-      @current_user = current_user
-      unless params[:food][:calories] =~ /\A\d+\Z/
-        flash[:food_create_error] = "* Enter a numerical value for your calorie intake."
+    if logged_in?
+      new_food = current_user.foods.new(params[:food])
+      unless new_food.valid?
+        flash[:food_create_error] = new_food.errors.details[:calories].any?{|detail| detail[:error] == :not_a_number} ? "* Enter a numerical value for your calorie intake." : "* Please fill out all fields."
         redirect '/foods/new'
-      end
-      new_achievement = Achievement.create(activity: @current_user.foods.create(params[:food]))
-      if new_achievement.valid?
-        redirect "/foods/users/#{@current_user.slug}"
       else
-        flash[:food_create_error] = "* Please fill out all fields."
-        redirect '/foods/new'
+        new_food.save
+        new_achievement = Achievement.create(activity: new_food)

```

Comment:
why are we creating a variable of new_achievement here? It seems to have no other use.